pkg/virt-launcher/virtwrap/cmd-server/server_test.go

```diff
@@ -118,17 +118,45 @@ var _ = Describe("Virt remote commands", func() {
 			Expect(err).ToNot(HaveOccurred())
 		})
 
-		It("should list domains", func() {
+		It("should list domains when no cache info exists", func() {
 			var list []*api.Domain
 			list = append(list, api.NewMinimalDomain("testvmi1"))
 
 			domainManager.EXPECT().ListAllDomains().Return(list, nil)
+			domainManager.EXPECT().GetGuestOSInfo().Return(nil)
+			domainManager.EXPECT().InterfacesStatus(list[0].Spec.Devices.Interfaces).Return(nil)
 			domain, exists, err := client.GetDomain()
 			Expect(err).ToNot(HaveOccurred())
 
 			Expect(exists).To(BeTrue())
 			Expect(domain).ToNot(Equal(nil))
 			Expect(domain.ObjectMeta.Name).To(Equal("testvmi1"))
+			Expect(domain.Status.OSInfo).To(Equal(api.GuestOSInfo{}))
+			Expect(domain.Status.Interfaces).To(BeNil())
+		})
+
+		It("should list domains when cache info exists", func() {
+			var list []*api.Domain
+			list = append(list, api.NewMinimalDomain("testvmi1"))
+
+			fakeInterfaces := []api.InterfaceStatus{
+				{
+					Name: "eth1",
+					Mac:  "00:00:00:00:00:01",
+				},
+			}
+			domainManager.EXPECT().InterfacesStatus(list[0].Spec.Devices.Interfaces).Return(fakeInterfaces)
+			domainManager.EXPECT().ListAllDomains().Return(list, nil)
+			domainManager.EXPECT().GetGuestOSInfo().Return(&api.GuestOSInfo{Name: "fedora"})
+
+			domain, exists, err := client.GetDomain()
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(exists).To(BeTrue())
+			Expect(domain).ToNot(Equal(nil))
```

Comment:
I do not know where from you took this assumption.
`BeNil` is checking nil, not an empty list or an empty anything.

Reply:
What you showed is wrong.
One `nil` is of a specific type and the other has no type. I do not even know how you managed to reach such a result.

Reply:
in case you have a function that returns type []api.InterfaceStatus` and actually return nil, this what will happen

Reply:
Per what I know, it will return a list type with a nil value, not a non-type nil like you claim.
It is true that you can compare any type nil value to a non-type nil, but that is surely an exception here to allow some level of generics.
I would conclude with a snippet:
```golang
It("foo", func() {
        var list []string
        Expect(list).To(BeNil())

        var list2 = []string{}
        Expect(list2).NotTo(BeNil())
})
```

Reply:
Try please `To(Equal(nil))` and see the assert that i shown